runtime/src/comm/gasnet/comm-gasnet-ex.c

```diff
@@ -946,8 +991,37 @@ int chpl_comm_run_in_lldb(int argc, char* argv[], int lldbArgnum, int* status) {
   return 0;
 }
 
+static void start_gasnet_progress_threads(void) {
+  unsigned int count;
+  const gex_ProgressThreadInfo_t *info;
+  GASNET_Safe(gex_System_QueryProgressThreads(myclient, &count, &info, 0) );
+  for (int i = 0; i < count; i++) {
+
+    // Don't create an internal receive thread if there is an external one
+    if (pollingRequired &&
+        (info[i].gex_thread_roles == GEX_THREAD_ROLE_RCV)) {
+      continue;
+    }
+    if (chpl_task_createCommTask((chpl_fn_p) info[i].gex_progress_fn,
+                                 info[i].gex_progress_arg, reservedCore)) {
+      chpl_internal_error("unable to start internal GASNet progress thread");
+    }
+  }
+}
+
 void chpl_comm_post_task_init(void) {
   start_polling();
+  start_gasnet_progress_threads();
+#if defined(GASNET_CONDUIT_IBV)
```

Comment:
@jhh67 I see a couple issues with the `GASNET_SND_THREAD` portion of the verbose output below which I think diminish its value.

1. If the GASNet-EX configure arguments disable the send thread, then reporting a true value of the env var gives the "wrong answer".
2. If I change the default value to enable the send thread by default in a future release, then reporting for the unset case will also give the "wrong answer".

Instead of the `enabled` here, I suggest a `have_snd_thread` Boolean which starts `false`, but is set to `true` if the loop above observes `info[i].gex_thread_roles && GEX_THREAD_ROLE_SND`

Reply:
Note: Paul meant `&` above not `&&`.

Maybe also add an analogous `printf` for the recv thread?